File: library/src/main/java/com/afollestad/aesthetic/AutoSwitchMode.java
package com.afollestad.aesthetic;

/** @author Aidan Follestad (afollestad) */
public enum AutoSwitchMode {
  OFF(0),
  ON(1),
  AUTO(2);

  private final int value;

  AutoSwitchMode(int value) {
    this.value = value;
  }

  public int toInt() {
    return value;
  }

  public static AutoSwitchMode fromInt(int value) {
    switch (value) {
      case 0:
        return OFF;
      case 1:
        return ON;
      default:
        return AUTO;
    }
  }
}

File: library/src/main/java/com/afollestad/aesthetic/AutoSwitchMode.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.aesthetic

/** @author Aidan Follestad (afollestad) */
enum class AutoSwitchMode(val value: Int) {
  OFF(0),
  ON(1),
  AUTO(2);

  companion object {

    fun fromInt(value: Int): AutoSwitchMode {
      return when (value) {
        0 -> OFF
        1 -> ON
        else -> AUTO
      }
    }
  }
}
